Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -1 +1,78 @@
 #include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/Utility.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+void LocalOrderImbalanceIndicatorModel::publish(
+    const OrderImbalance& imbalance) {
+  insert_sorted(imbalance);
+  m_imbalance_published_signal(imbalance);
+}
+
+void LocalOrderImbalanceIndicatorModel::insert(
+    const OrderImbalance& imbalance) {
+  insert_sorted(imbalance);
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    LocalOrderImbalanceIndicatorModel::load(const TimeInterval& interval) {
+  auto first = std::vector<OrderImbalance>::iterator();
```

Comment:
Initialize your variables at the point of declaration.

    auto first = [&] {
      if(boost::icl::is_left_closed(interval.bounds())) {
        return std::lower_bound(m_imbalances.begin(), m_imbalances.end(),
          interval.lower(), [] (const auto& imbalance, const auto& timestamp) {
            return imbalance.m_timestamp < timestamp;
          });
      } else {
        return std::upper_bound(m_imbalances.begin(), m_imbalances.end(),
          interval.lower(), [] (const auto& timestamp, const auto& imbalance) {
            return imbalance.m_timestamp > timestamp;
        });
      }
    }();